api/v1/services/CheckInService.js

```diff
@@ -62,6 +87,35 @@ module.exports.createCheckIn = function (attributes){
                 var source = "userId";
                 throw new errors.InvalidParameterError(message, source);
             }
-            return checkin.save();
+            return CheckIn.transaction(function (t) {
+                return checkin.save(null, {transacting: t})
+                .then(function(model){
+                    if(credentialsRequested){
+                        return NetworkCredential.findUnassigned()
+                        .then(function(networkCredential){
```

Comment:
Please do not nest promises like this. Return the result of `findUnassigned` and pick it up in the next section of the promise chain. That goes for the remainder of this too.